pkg/apis/azure/helper/scheme.go

```diff
@@ -163,3 +163,15 @@ func InfrastructureStatusFromInfrastructure(infra *extensionsv1alpha1.Infrastruc
 	}
 	return status, nil
 }
+
+// WorkloadIdentityConfigFromBytes extracts WorkloadIdentityConfig from the provided byte array.
+func WorkloadIdentityConfigFromBytes(config []byte) (*api.WorkloadIdentityConfig, error) {
+	if len(config) == 0 {
+		return nil, fmt.Errorf("cannot parse WorkloadIdentityConfig from empty config")
+	}
+	workloadIdentityConfig := &api.WorkloadIdentityConfig{}
+	if _, _, err := decoder.Decode(config, nil, workloadIdentityConfig); err != nil {
```

Comment:
Should we use util.Decode here?

```
"github.com/gardener/gardener/extensions/pkg/util"
```

Reply:
> We already agreed on this approach in provider-aws

That change I did not see 🙈 

> Is there any reason to use it? 

I guess the benefit is this comment https://github.com/gardener/gardener/blob/56037371cdd961ee894592c6bcaf101d0aabe9bc/extensions/pkg/util/serialization.go#L17-L19

TBH I see that both are being used through the repository and I am not sure when to use which. My guess is that util.Decode is used during validation to ensure proper api versioning. The contra question would be "Why introduce the change in the first place? What was the benefit?"